src/OpenTK/Platform/MacOS/CocoaNativeWindow.cs

```diff
@@ -678,6 +680,8 @@ public override void ProcessEvents()
                     case NSEventType.FlagsChanged:
                         {
                             var modifierFlags = (NSEventModifierMask)Cocoa.SendUint(e, selModifierFlags);
+                            capslockOn = (modifierFlags & NSEventModifierMask.AlphaShiftKeyMask) != 0;
```

Comment:
If you're going to work with modifier flags, you should put it in `UpdateModifierFlags`.